internal/querycoordv2/meta/replica_manager.go

```diff
@@ -30,6 +30,36 @@ import (
 	"github.com/milvus-io/milvus/pkg/util/typeutil"
 )
 
+// Deprecated: only for test, remove in future.
+// NewReplicaForPlanAtDefaultRG creates a new ReplicaForPlan.
+func NewReplicaForPlanAtDefaultRG(replicaID int64) *ReplicaForPlan {
+	return &ReplicaForPlan{
+		replicaID:     replicaID,
+		resourceGroup: DefaultResourceGroupName,
+	}
+}
+
+// Deprecated: ReplicaForPlan is a replica for querycoord balance plan.
+// TODO: should be same with Replica, but it has lock now, remove this struct after resource group enhancement.
+type ReplicaForPlan struct {
```

Comment:
Will be removed by #30564

Reply:
Maybe `Replica` will do for meta info? Since the mutex is used for nodes information update

Reply:
`ResourceGroup` field in `Replica` struct may be updated. 
We may be sent an updated latest `ResourceGroup` field to wrong queryNode.
`Replica` here should be a snapshot of Replica Meta in query coord.
After ResourceGroup  enhancement, I will remove these `ReplicaForPlan` struct, use `copy on write` type to maintain Replica struct without any updates.

Reply:
> `ResourceGroup` field in `Replica` struct may be updated. We may be sent an updated latest `ResourceGroup` field to wrong queryNode. `Replica` here should be a snapshot of Replica Meta in query coord. After ResourceGroup enhancement, I will remove these `ReplicaForPlan` struct, use `copy on write` type to maintain Replica struct without any updates.

ReplicaForPlan cannot handle the problem, I will use Replica to replice it. wait for Resource Group Enhancement to handle it.

Reply:
Indeed, the copy here cannot solve the problem.
Actually, the load meta can not be accurate as the resource group might be updated after the load request is dispatched to querynodes?

Reply:
We need a consistent view of replica when creating balance task.
Otherwise, wrong resource group will be assigned to query node.
Finally, it can be updated correctly, but wrong metric is generated.